playbook/app/pb_kits/playbook/pb_icon/icon.rb

```diff
@@ -79,19 +79,42 @@ def custom_icon_classname
         )
       end
 
-      def render_svg(path)
-        if File.extname(path) == ".svg"
-          doc = Nokogiri::XML(URI.open(path)) # rubocop:disable Security/Open
-          svg = doc.at_css "svg"
-          svg["class"] = "pb_custom_icon " + object.custom_icon_classname
-          raw doc
-        else
-          raise("Custom icon must be an svg. Please check your path and file type.")
-        end
+      def render_svg
+        doc = Nokogiri::XML(URI.open(icon || custom_icon)) # rubocop:disable Security/Open
+        svg = doc.at_css "svg"
+        svg["class"] = "pb_custom_icon " + object.custom_icon_classname
+        svg["height"] = svg_dims[svg_size] * 16
+        svg["width"] = svg_dims[svg_size] * 16
+        doc.at_css("path")["fill"] = "currentColor"
+        raw doc
+      end
+
+      def is_svg?
+        (icon || custom_icon.to_s).include?(".svg")
       end
 
     private
 
+      def svg_size
+        size.nil? ? "1x" : size
+      end
+
+      def svg_dims
+        { "lg" => 1.25,
+          "xs" => 0.75,
+          "sm" => 0.875,
+          "1x" => 1,
+          "2x" => 2,
+          "3x" => 3,
+          "4x" => 4,
+          "5x" => 5,
+          "6x" => 6,
+          "7x" => 7,
+          "8x" => 8,
+          "9x" => 9,
+          "10x" => 10 }
+      end
```

Comment:
I think this should work how it does now. Just allowing the classes and let our fa.css size this correctly.

Reply:
This is removed, we just need to set the `width="auto"` and `height="auto"` on the svg to get the FA styles to apply size correctly